src/Sarif.Viewer.VisualStudio.Core/Models/ScrollViewerWrapper.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text.Adornments;
+using Microsoft.VisualStudio.Text.Tagging;
+
+using static System.Windows.Forms.VisualStyles.VisualStyleElement;
+
+namespace Sarif.Viewer.VisualStudio.Core.Models
+{
+    /// <summary>
+    /// Wraps other render-able content in a Scroll viewer, preventing content from going off screen when possible.
+    /// </summary>
+    internal class ScrollViewerWrapper : IErrorTag
+    {
+        /// <summary>
+        /// A collection of error tags that are to be wrapped and displayed in order.
+        /// </summary>
+        private readonly List<IErrorTag> objectsToWrap;
+
+        private static readonly Dictionary<string, int> errorTypeToRank = new Dictionary<string, int>()
+        {
+            { PredefinedErrorTypeNames.SyntaxError, 1 },
+            { PredefinedErrorTypeNames.CompilerError, 2 },
+            { PredefinedErrorTypeNames.OtherError, 3 },
+            { PredefinedErrorTypeNames.Warning, 4 },
+            { PredefinedErrorTypeNames.Suggestion, 5 },
+            { PredefinedErrorTypeNames.HintedSuggestion, 6 },
+        };
+
+        public ScrollViewerWrapper(List<IErrorTag> objectsToWrap)
+        {
+            this.objectsToWrap = objectsToWrap;
+        }
+
+        /// <summary>
+        /// Gets the highest error type of the error tags that are wrapped.
+        /// The order of severity is defined in <see cref="ScrollViewerWrapper.errorTypeToRank"/>.
+        /// </summary>
+        public string ErrorType
+        {
+            get
+            {
+                if (this.objectsToWrap.Count == 0)
+                {
+                    return string.Empty;
+                }
+
+                string highestErrorStr = PredefinedErrorTypeNames.HintedSuggestion;
+                int highestRank = errorTypeToRank[highestErrorStr];
+                foreach (IErrorTag errorTag in this.objectsToWrap)
+                {
+                    string errorType = errorTag.ErrorType;
+                    int tagRank = 1;
+                    if (errorTypeToRank.ContainsKey(errorType))
+                    {
+                        tagRank = errorTypeToRank[errorType];
+                    }
+                    else
+                    {
+                        tagRank = errorTypeToRank[PredefinedErrorTypeNames.OtherError];
+                    }
+
+                    if (tagRank < highestRank)
+                    {
+                        highestRank = tagRank;
+                        highestErrorStr = errorType;
+                    }
+                }
+
+                return highestErrorStr;
+            }
+        }
+
+        /// <summary>
+        /// Gets the content that is to be displayed to the end user. Will wrap all of the child content in a scroll viewer, and will scale the height to prevent content from goin off screen.
+        /// </summary>
+        public object ToolTipContent
```

Comment:
>object [](http://example.com/codeflow?start=15&length=6)

Does this need to return object, or can it be ScrollViewer?

Reply:
The IErrorTag interface requires a ToolTipContent of object, 
<img width="933" alt="image" src="https://user-images.githubusercontent.com/126511432/226977871-bd0c9e1f-c1a0-4e56-ba43-6d7a298139c4.png">
I could do a 
public ScrollViewer ToolTipContent and a public object TooltipContent and have one call the other but you do public object ToolTipContent for SarifLocationErrorTag too so I figured it would be best to keep as-is